### PR TITLE
volume: use generic sets instead of specified sets

### PR DESCRIPTION
**Pull request by bzsuni**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
specific sets are deprecated
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
none
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
none
```

### CODE REVIEW

File: pkg/volume/util/atomic_writer.go [hunk 1]
```diff
@@ -146,30 +146,30 @@ func (w *AtomicWriter) Write(payload map[string]FileProjection, setPerms func(su
 
 	// (2)
 	dataDirPath := filepath.Join(w.targetDir, dataDirName)
-	oldTsDir, err := os.Readlink(dataDirPath)
+	oldTSDir, err := os.Readlink(dataDirPath)
```
Comment (haircommander) [inline pkg/volume/util/atomic_writer.go:149]: I would say technically this belongs in its own commit or PR

Comment (bzsuni) [reply pkg/volume/util/atomic_writer.go:149]: Yes, you are right.
Because the lint check reported an error, I made modifications here.
I will resubmit a PR about this. Thanks very much @haircommander

**Comment by bzsuni** [comment]
Hi @dims I have updated the code. Can you help me remove the ‘rotten’ label and review it for me?

File: pkg/volume/util/atomic_writer.go (hunk 1 above)
Comment (bart0sh) [reply pkg/volume/util/atomic_writer.go:149]: @bzsuni please address the comment above and I'll lgtm this PR.

Comment (bzsuni) [reply pkg/volume/util/atomic_writer.go:149]: If the spelling is not changed, CI check will report an error. 
Should I change the spelling in this PR, or in a separate PR?
@bart0sh

Comment (bart0sh) [reply pkg/volume/util/atomic_writer.go:149]: Let's move everything unrelated to sets into a separate commit and I think this should be enough.

Comment (bzsuni) [reply pkg/volume/util/atomic_writer.go]: I have moved the spelling check to #120876
Could you help me lgtm @bart0sh

**Comment by bzsuni** [comment]
/assign @bart0sh @xing-yang 

for approval please

**Review by bart0sh — APPROVED** [review]
/lgtm

**Comment by bzsuni** [comment]
/assign @yujuhong 

Could you please review this PR?

**Comment by bart0sh** [comment]
@bzsuni please, rebase, thanks!

**Comment by bzsuni** [comment]
/assign @thockin 
For approval please

**Comment by thockin** [comment]
I don't understand what this PR is doing - these changes should not be needed.  Am I missing something?

**Comment by bzsuni** [comment]
> I don't understand what this PR is doing - these changes should not be needed.

Yes, these changes are not needed. Thank you very much.
I didn’t check the code after rebase. Sorry about that.
